streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -250,14 +250,10 @@ public void completeRestoration() {
     public void suspend() {
         switch (state()) {
             case CREATED:
-            case SUSPENDED:
```

Comment:
IMHO, we should keep the `SUSPENDED` case for consistency reasons. Only merge `CREATED` and `RESTORING` (cf. my other comment on `Task.java`)